test/langtools/jdk/javadoc/doclet/testIndexInherited/TestIndexInherited.java

```diff
@@ -0,0 +1,159 @@
+        {"p":"pkg2","c":"ClassC","l":"methodA()"}""", """
+        {"p":"pkg1","c":"ClassB","l":"STRING_A"}""", """
+        {"p":"pkg2","c":"ClassC","l":"STRING_A"}"""};
+
+    /**
+     * Index entries for members declared in the superclass.
+     */
+    private static final String[] INDEX_SUPERCLASS = {"""
+        <dt><a href="pkg1/ClassA.html#methodA()" class="member-name-link">methodA()</a> \
+        - Method in interface pkg1.<a href="pkg1/ClassA.html" title="interface in pkg1">ClassA</a></dt>
+        """, """
+        <dt><a href="pkg1/ClassA.html#STRING_A" class="member-name-link">STRING_A</a> \
+        - Static variable in interface pkg1.<a href="pkg1/ClassA.html" title="interface in pkg1">ClassA</a></dt>
+        """};
+
+    /**
+     * Search entries for members declared in the superclass.
+     */
+    private static final String[] SEARCH_SUPERCLASS = {"""
+        {"p":"pkg1","c":"ClassA","l":"methodA()"}""", """
+        {"p":"pkg1","c":"ClassA","l":"STRING_A"}"""};
+
+    /**
+     * The default constructor.
+     */
+    public TestIndexInherited() {
+    }
+
+    /**
+     * Runs the test methods.
+     *
+     * @param args the command-line arguments
+     * @throws Exception if an errors occurs while executing a test method
+     */
+    public static void main(String... args) throws Exception {
+        TestIndexInherited tester = new TestIndexInherited();
+        tester.runTests(m -> new Object[]{Path.of(m.getName())});
+    }
+
+    /**
+     * Tests entries in the subclasses, loaded in alphabetical order.
+     *
+     * @param base the base directory for test output
+     */
+    @Test
+    public void testSubclasses1(Path base) {
+        String dir = base.resolve("out").toString();
+        javadoc("-d", dir, "-sourcepath", testSrc, "pkg1", "pkg2");
+        checkExit(Exit.OK);
+        checkOrder(INDEX_FILE, INDEX_SUBCLASSES);
+        checkOrder(SEARCH_FILE, SEARCH_SUBCLASSES);
+        checkOutput(INDEX_FILE, false, "ClassA");
+        checkOutput(SEARCH_FILE, false, "ClassA");
+    }
+
+    /**
+     * Tests entries in the subclasses, loaded in reverse alphabetical order.
+     *
+     * @param base the base directory for test output
+     */
+    @Test
+    public void testSubclasses2(Path base) {
+        String dir = base.resolve("out").toString();
+        javadoc("-d", dir, "-sourcepath", testSrc, "pkg2", "pkg1");
+        checkExit(Exit.OK);
+        checkOrder(INDEX_FILE, INDEX_SUBCLASSES);
+        checkOrder(SEARCH_FILE, SEARCH_SUBCLASSES);
+        checkOutput(INDEX_FILE, false, "ClassA");
+        checkOutput(SEARCH_FILE, false, "ClassA");
+    }
+
+    /**
+     * Tests entries in the superclass.
+     *
+     * @param base the base directory for test output
+     */
+    @Test
+    public void testSuperclass(Path base) {
+        String dir = base.resolve("out").toString();
+        javadoc("-d", dir, "-sourcepath", testSrc, "-private", "pkg1", "pkg2");
```

Comment:
At first glance, the use of the `-private` option does not seem to match the method name `testSuperclass` or its matching description (tests entries in the superclass).

There are two characteristics here that seem to be getting mixed up:

1. testing subclasses and superclasses
2. testing with default access (document public and protected) and with "more" access, such as using `-package` or `-private` options.

Ideally, the tests should test subclasses and superclasses with default access, and then repeat, testing subclasses and superclasses with "more" access, such as with `-private`.

Reply:
Right. I renamed most of the members and methods of the test class and rewrote its comments, which were far too terse. I also added a check that the inherited members do not show up in the index when they're already listed for the declaring superclass. I think it's much better now, but let me know what you think.